nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/ScrollElasticsearchHttp.java

```diff
@@ -0,0 +1,415 @@
+                final long startNanos = System.nanoTime();
+
+                final Response getResponse = sendRequestToElasticsearch(okHttpClient, queryUrl,
+                        username, password, "GET", null);
+                this.getPage(getResponse, queryUrl, context, session, flowFile, logger, startNanos);
+            }
+
+        } catch (IOException ioe) {
+            logger.error(
+                    "Failed to read from Elasticsearch due to {}, this may indicate an error in configuration "
+                            + "(hosts, username/password, etc.).",
+                    new Object[] { ioe.getLocalizedMessage() }, ioe);
+            session.remove(flowFile);
+            context.yield();
+
+        } catch (Exception e) {
+            logger.error("Failed to read {} from Elasticsearch due to {}", new Object[] { flowFile,
+                    e.getLocalizedMessage() }, e);
+            session.transfer(flowFile, REL_FAILURE);
+            context.yield();
+        }
+    }
+
+    private void getPage(final Response getResponse, final URL url, final ProcessContext context,
+            final ProcessSession session, FlowFile flowFile, final ComponentLog logger, final long startNanos)
+            throws IOException {
+        final int statusCode = getResponse.code();
+
+        if (isSuccess(statusCode)) {
+            ResponseBody body = getResponse.body();
+            final byte[] bodyBytes = body.bytes();
+            JsonNode responseJson = parseJsonResponse(new ByteArrayInputStream(bodyBytes));
+            String scrollId = responseJson.get("_scroll_id").asText();
+
+            StringBuilder builder = new StringBuilder();
+
+            builder.append("{ \"hits\" : [");
+
+            JsonNode hits = responseJson.get("hits").get("hits");
+            if (hits.size() == 0) {
+                finishQuery(context.getStateManager());
+                session.remove(flowFile);
+                return;
+            }
+
+            for(int i = 0; i < hits.size(); i++) {
+                JsonNode hit = hits.get(i);
+                String retrievedIndex = hit.get("_index").asText();
+                String retrievedType = hit.get("_type").asText();
+
+                JsonNode source = hit.get("_source");
+                flowFile = session.putAttribute(flowFile, "es.index", retrievedIndex);
+                flowFile = session.putAttribute(flowFile, "es.type", retrievedType);
+
+                builder.append(source.toString());
+                if (i < hits.size() - 1) {
+                    builder.append(", ");
+                }
+            }
+            builder.append("] }");
+            logger.debug("Elasticsearch retrieved " + responseJson.size() + " documents, routing to success");
+
+            flowFile = session.write(flowFile, out -> {
+                out.write(builder.toString().getBytes());
+            });
+            session.transfer(flowFile, REL_SUCCESS);
```

Comment:
Since the content of the flow file is a JSON object, I recommend setting the mime.type attribute to application/json.